tracer/src/Datadog.Trace/HttpHeaderNames.cs

```diff
@@ -25,6 +25,18 @@ public static class HttpHeaderNames
         /// </summary>
         public const string SamplingPriority = "x-datadog-sampling-priority";
 
+        /// <summary>
+        /// Used to propagate sampling mechanism across tracer versions in the same process,
+        /// not for propagation across process boundaries.
+        /// </summary>
+        internal const string SamplingMechanism = "x-datadog-sampling-mechanism";
```

Comment:
I think we should come up with special internal keys for internal propagation. The keys used in headers need to be switchable/customizable (e.g. Datadog headers vs B3 headers), but these are no included there.